hera_pspec/uvwindow.py

```diff
@@ -754,8 +827,8 @@ def get_cylindrical_wf(self, bl_len,
         else:
             self.check_kunits(kperp_bins)
         kperp_bins = np.array(kperp_bins.value)
-        if not np.isclose(np.diff(kperp_bins),np.diff(kperp_bins)[0]).all():
-            warnings.warn('kperp_bins must be linearly spaced.')
+        if not np.isclose(np.diff(kperp_bins), np.diff(kperp_bins)[0]).all():
+            warnings.warn('get_cylindrical_wf: kperp_bins must be linearly spaced.')
```

Comment:
why does this have to be true? And if so, why is it just a warning, not an error?

Reply:
That is because the edges of each bin are obtained assuming a linear spacing, e.g.:
`kperp_bin_edges = np.arange(kperp_bins.min()-dk_perp/2, kperp_bins.max() + dk_perp, step=dk_perp)`.
It can be modified in the future but I don't think handling log-spaced bins is necessary at the moment (the data is linearly spaced).

Reply:
Yes I changed for a `ValueError`! Sorry, forgot to mention

Reply:
👍🏻